subprojects/docs/src/docs/userguide/authoring-builds/introduction/plugins.adoc

```diff
@@ -0,0 +1,579 @@
+
+Plugins can:
+
+* Extend the Gradle model (e.g., add new DSL elements that can be configured).
+* Configure the project according to conventions (e.g., add new tasks or configure sensible defaults).
+* Apply specific configuration (e.g., add organizational repositories or enforce standards).
+
+There are many advantages to applying plugins over adding logic to the project build script:
+
+* Promotes reuse and reduces the overhead of maintaining similar logic across multiple projects.
+* Allows a higher degree of modularization, enhancing comprehensibility and organization.
+* Encapsulates imperative logic and allows build scripts to be as declarative as possible.
+
+== Plugin distribution
+
+Plugins are available in three ways:
+
+1. **Core plugins** - Gradle develops and maintains a set of <<plugin_reference#plugin_reference,Core Plugins>>.
+2. **Community plugins** - Gradle plugins shared in a remote repository such as Maven or the https://plugins.gradle.org[Gradle Plugin Portal].
+3. **Local plugins** - Gradle enables users to create *custom* plugins using link:{groovyDslPath}/org.gradle.api.tasks.javadoc.Javadoc.html[APIs].
+
+[[sec:types_of_plugins]]
+== Types of plugins
+
+There are two general types of plugins in Gradle: *binary* plugins and *script* plugins.
+
+*Binary plugins* are written either programmatically by implementing the link:{javadocPath}/org/gradle/api/Plugin.html[`Plugin`] interface or through declarations in either Groovy or Kotlin DSL.
+They can reside within a build script, the project hierarchy, or externally in a plugin jar.
+
+*Script plugins* are additional build scripts that further configure the build and usually implement a declarative approach to manipulating the build.
+They are typically used within a build but can be externalized and accessed remotely.
+
+A plugin often starts as a script plugin (because they are easy to write).
+Then, as the code becomes more valuable, it's migrated to a binary plugin that can be easily tested and shared between multiple projects or organizations.
+
+[[sec:using_plugins]]
+== Using plugins
+
+To use the build logic encapsulated in a plugin, Gradle needs to perform two steps.
+First, it needs to *resolve* the plugin, and then it needs to *apply* the plugin to the target, usually a link:{groovyDslPath}/org.gradle.api.Project.html[`Project`].
+
+*Resolving* a plugin means finding the correct version of the jar that contains a given plugin and adding it to the script classpath.
+Once a plugin is resolved, its API can be used in a build script.
+Script plugins are self-resolving in that they are resolved from the specific file path or URL provided when applying them.
+Core binary plugins provided as part of the Gradle distribution are automatically resolved.
+
+*Applying* a plugin means executing the plugin's link:{javadocPath}/org/gradle/api/Plugin.html#apply-T-[Plugin.apply(T)] on the Project you want to enhance with the plugin.
+Applying plugins is _idempotent_.
```

Comment:
💅 Should be idempotent.  Core Gradle plugins are, but this isn't something that happens automatically, you need to make this the case when writing your own custom plugins.